src/mame/drivers/albazc.cpp

```diff
@@ -207,12 +207,12 @@ static INPUT_PORTS_START( hanaroku )
 	PORT_BIT( 0x80, IP_ACTIVE_LOW, IPT_HANAFUDA_NO )
 
 	PORT_START("IN2")   /* 0xe002 */
-	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_SERVICE3 ) PORT_NAME("Data Clear")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_SERVICE4 ) PORT_NAME("Data Clear")
 	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_TILT )
-	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON9 ) PORT_NAME("Medal In") PORT_CODE(KEYCODE_I)
-	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_BUTTON10 ) PORT_NAME("Pay Out") PORT_CODE(KEYCODE_O)
-	PORT_BIT( 0x10, IP_ACTIVE_LOW, IPT_SERVICE1 ) PORT_NAME("Ext In 1")
-	PORT_BIT( 0x20, IP_ACTIVE_LOW, IPT_SERVICE2 ) PORT_NAME("Ext In 2")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_MAHJONG_I ) PORT_NAME("Medal In")
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_MAHJONG_O ) PORT_NAME("Pay Out")
```

Comment:
I take that back (Using IPT_OTHER) - Your suggestion sounds more sensible.
I will change them to IPT_GAMBLE_KEYIN & IPT_GAMBLE_KEYOUT.
How they work - That's something else.